test/gtest/group_conv3d_bwd.hpp

```diff
@@ -0,0 +1,203 @@
+                  << " pad_y:" << tc.pad_y << " pad_x:" << tc.pad_x << " stride_z:" << tc.stride_z
+                  << " stride_y:" << tc.stride_y << " stride_x:" << tc.stride_x
+                  << " dilation_z:" << tc.dilation_z << " dilation_y:" << tc.dilation_y
+                  << " dilation_x:" << tc.dilation_x << " conv_mode:" << tc.conv_mode;
+    }
+
+    std::vector<size_t> GetInput() { return {N, C, D, H, W}; }
+    std::vector<size_t> GetWeights() { return {k, C, z, y, x}; }
+
+    miopen::ConvolutionDescriptor GetConv()
+    {
+        return miopen::ConvolutionDescriptor{
+            3,
+            miopenConvolution,
+            miopenPaddingDefault,
+            {static_cast<int>(pad_z), static_cast<int>(pad_y), static_cast<int>(pad_x)},
+            {static_cast<int>(stride_z), static_cast<int>(stride_y), static_cast<int>(stride_x)},
+            {static_cast<int>(dilation_z),
+             static_cast<int>(dilation_y),
+             static_cast<int>(dilation_x)},
+            {0, 0, 0},
+            static_cast<int>(G),
+            1.0};
+    }
+};
+
+std::vector<ConvTestCase> ConvTestConfigs()
+{ // g    n   c   d    h   w   k   z  y  x pad_x pad_y pad_z stri_x stri_y stri_z dia_x dia_y dia_z
+    return {{1, 128, 64, 14, 28, 28, 64, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {1, 64, 32, 28, 28, 28, 32, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {32, 128, 32, 28, 28, 28, 32, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {16, 128, 16, 28, 28, 28, 16, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {8, 128, 8, 28, 28, 28, 8, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {4, 128, 4, 28, 28, 28, 4, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {2, 128, 2, 28, 28, 28, 2, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution}};
+}
+
+inline int SetTensorLayout(miopen::TensorDescriptor& desc)
+{
+    // get layout string names
+    std::string layout_str = desc.GetLayout_str();
+
+    std::vector<std::size_t> lens = desc.GetLengths();
+    std::vector<int> int_lens(lens.begin(), lens.end());
+
+    // set the strides for the tensor
+    return SetTensorNd(&desc, int_lens, layout_str, desc.GetType());
+}
+
+template <typename T = float>
+struct ConvBwdSolverTest
+    : public ::testing::TestWithParam<
+          std::tuple<miopenConvBwdDataAlgorithm_t, ConvTestCase, miopenTensorLayout_t>>
+{
+protected:
+    void SetUp() override
+    {
+        test_skipped = false;
+
+        std::tie(algo, conv_config, tensor_layout) = GetParam();
+        input   = tensor<T>{miopen_type<T>{}, tensor_layout, conv_config.GetInput()};
+        weights = tensor<T>{miopen_type<T>{}, tensor_layout, conv_config.GetWeights()};
+        SetTensorLayout(input.desc);
+        SetTensorLayout(weights.desc);
+        std::random_device rd{};
+        std::mt19937 gen{rd()};
```

Comment:
I know it's copy-pasted from the other convolutions tests, but:
1) `mt19937` is slow
2) random device make the test unreproducible from run to run

Reply:
Unproducible but also unpredictable,  we want the input somehow unpredictable because we do not know what input the users will use. So random number should be fine here for testing purpose.

Reply:
No. It means that we can have random CI failures. And, the most important and critical thing - we will NOT have a way to reproduce and debug the issue because we have no ability to generate exact data which lead to that failure. It's essential verification practice.

If you need a randomness it must be special set of tests which:
1) do not block CI in case of failed tests
2) have an ability to reproduce exact random sequence

So random seed should be removed. Or there must be a way to report and set this random seed.

Reply:
Another point - we've already gotten standardized PRNG facility and the better way is to use that facility.